Snowflake.Data/Core/Session/ConnectionPoolManager.cs

```diff
@@ -102,6 +102,9 @@ public int GetCurrentPoolSize()
 
         public bool SetPooling(bool poolingEnabled)
         {
+            if (!poolingEnabled)
+                throw new Exception(
```

Comment:
why? if an app has several pools and wants to switch them off in one shot - why not allow this?

Reply:
due to our conversation. You wanted in the next steps to introduce parameter in the connection string to control if the pool is to use. It would be the way to control the pool usage by the user. We can discuss it.